### PR TITLE
Support Sparse Tensor Constant

### PR DESCRIPTION
**Pull request by KKimj**
## Support Sparse Tensor Constant

It works with ```tensor.extract``` as well.

#### encode ```ConstantOp```
1. check the ```ConstantOp op``` isa ```SparseElementsAttr```
2. check whether  Type of elements, ```eltType``` is ```FloatType``` or ```IntegerType```
    - else ;  ```unsupported type```
3. make Tensor  as invoking ```st.regs.add```  with ```vector``` of ```z3::expr``` and ```dimvec```

### CODE REVIEW

**Comment by KKimj** [comment]
Should I write docstring at that code?

**Comment by makesource** [comment]
Could you add simple litmus tests that can check correctness of sparse encoding?

**Comment by KKimj** [comment]
@makesource 
I uploaded tests and resolved it! Thanks!

**Comment by KKimj** [comment]
Spec: https://mlir.llvm.org/docs/Dialects/Builtin/#sparseelementsattr

File: src/vcgen.cpp [hunk 1]
```diff
@@ -670,6 +670,34 @@ optional<string> encodeOp(State &st, mlir::ConstantOp op) {
 
     st.regs.add(op, Integer(i.getSExtValue(), bw));
     return {};
+  } else if (mlir::SparseElementsAttr sparseAttr = attr.dyn_cast<mlir::SparseElementsAttr>()) {
+    std::vector<z3::expr> sparse_values;
```
Comment (makesource) [inline src/vcgen.cpp:674]: I think it's good to follow the same c++ naming convention as MLIR code. (i.e use camelcase rather than snake case)

Comment (KKimj) [reply src/vcgen.cpp:674]: Okay~

Comment (KKimj) [reply src/vcgen.cpp]: Please check!
I resolved it.
Thanks!

**Pull request merged**